src/include/platform/ConfigurationManager.h

```diff
@@ -102,8 +102,8 @@ class ConfigurationManager
     virtual CHIP_ERROR StoreRebootCount(uint32_t rebootCount)                          = 0;
     virtual CHIP_ERROR GetTotalOperationalHours(uint32_t & totalOperationalHours)      = 0;
     virtual CHIP_ERROR StoreTotalOperationalHours(uint32_t totalOperationalHours)      = 0;
-    virtual CHIP_ERROR GetBootReasons(uint32_t & bootReasons)                          = 0;
-    virtual CHIP_ERROR StoreBootReasons(uint32_t bootReasons)                          = 0;
+    virtual CHIP_ERROR GetBootReason(uint32_t & bootReason)                            = 0;
+    virtual CHIP_ERROR StoreBootReason(uint32_t bootReason)                            = 0;
```

Comment:
These setters should not be COnfigurationManager anymore, only on DiagnosticsDataProvider. Otherwise we have an extra dependency.

This can be refactored later.

Reply:
Those values, such as BootReason could be set before the device get rebooted, and this value need to survive through the power cycle in this case. Any values need to survive through the power cycle are currently stored in ConfigurationMgr